server/src/test/java/keywhiz/service/permission/AlwaysAllowDelegatingPermissionCheckTest.java

```diff
@@ -0,0 +1,71 @@
+package keywhiz.service.permission;
+
+import com.codahale.metrics.MetricRegistry;
+import java.util.Objects;
+import keywhiz.service.permissions.Action;
+import keywhiz.service.permissions.AlwaysAllowDelegatingPermissionCheck;
+import keywhiz.service.permissions.KeywhizPrincipal;
+import keywhiz.service.permissions.PermissionCheckMetricsNames;
+import keywhiz.service.permissions.PermissionCheck;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+public class AlwaysAllowDelegatingPermissionCheckTest {
+
+  @Rule public MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock PermissionCheck delegate;
+
+  MetricRegistry metricRegistry;
+  AlwaysAllowDelegatingPermissionCheck alwaysAllow;
+  KeywhizPrincipal principal;
+  Objects target;
+
+  @Before
+  public void setUp() {
+    metricRegistry = new MetricRegistry();
+    alwaysAllow = new AlwaysAllowDelegatingPermissionCheck(metricRegistry, delegate);
+  }
+
+  @Test public void delegateIsAllowedReturnsTrue() {
+    when(delegate.isAllowed(any(), any(), any())).thenReturn(true);
+
+    boolean permitted = alwaysAllow.isAllowed(principal, Action.ADD, target);
+
+    assertThat(permitted);
+    assertThat(metricRegistry.histogram(PermissionCheckMetricsNames.PermissionCheckHistogram).getCount()).isEqualTo(1);
+    assertThat(metricRegistry.histogram(PermissionCheckMetricsNames.PermissionCheckHistogram).getSnapshot().getMean()).isEqualTo(1);
+  }
+
+  @Test public void delegateIsAllowedReturnsFalse() {
+    when(delegate.isAllowed(any(), any(), any())).thenReturn(false);
```

Comment:
Totally a style thing, no action needed on your part:

Personally, I find mock programming to be pretty noisy, and I think all the when/any/thenReturn etc. can obscure the underlying intent or behavior. In this case, I might replace this line with a helper method like:

```
isAllowedReturns(true);
```

Short and sweet, without all the Mockito syntax cruft, and (I think) the behavior is totally obvious. Or, maybe we should throw a "delegate" in there too.

Same goes for other stuff in tests, too: if you have a ton of mostly-identical setup/execution code that only varies in a couple different ways, a helper method can keep the top-level test code concise and readable. My experience is that a lot of people think of testing code as a sort of second-class citizen where you don't need to be as tidy as actual application code, but I believe tests should be held to the same readability standards since they're essentially documentation-as-code.

Speaking of which, I like to use long test names that include not just the "situation" (as you've done here) but also the expected behavior. Something like "isAllowedReturnsTrueWhenDelegateReturnsFalse" is very literal and verbose, but something like "allowsWhenDelegateDenies" sounds a bit more human.